External/Plugins/CodeFormatter/Dialogs/HaxeAStyleDialog.cs

```diff
@@ -0,0 +1,427 @@
+﻿using CodeFormatter.Preferences;
+using CodeFormatter.Utilities;
+using PluginCore;
+using ScintillaNet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Forms;
+using PluginCore.Localization;
+
+namespace CodeFormatter.Dialogs
+{
+    public partial class HaxeAStyleDialog : Form
+    {
+        private ScintillaControl txtExample;
+
+        /// <summary>
+        /// Contains a mapping of bracket styles with the corresponding option for AStyle
+        /// </summary>
+        private static Dictionary<string, string> bracketStyles = new Dictionary<string, string>
+        {
+            { "Allman", "allman" },
+            { "Java", "java" },
+            { "Kernighan & Ritchie", "kr" },
+            { "Stroustrup", "stroustrup" },
+            { "Whitesmith", "whitesmith" },
+            { "VTK", "vtk" },
+            { "Banner", "banner" },
+            { "GNU", "gnu" },
+            { "Linux", "linux" },
+            { "Horstmann", "horstmann" },
+            { "One True Brace", "otbs" },
+            { "Google", "google" },
+            //{ "Mozilla", "mozilla" }, //not supported by old version of AStyle
+            { "Pico", "pico" },
+            { "Lisp", "lisp" },
+        };
```

Comment:
I'd say try to avoid this design pattern. It just keeps the dictionary permanent in the memory during the lifetime of the program.
Using `switch` statement is much better. The compiler decides whether to use `if-else` statements, or a one-time dictionary, depending on which it believes is faster.
Even a manual one-time dictionary every time the mapping is needed is still better, since instantiation doesn't cost much, and is better than having dictionaries stuck in the memory.